strangler-infusereo/code/ruby-service.rb

```diff
@@ -0,0 +1,37 @@
+require 'sinatra'
+require 'json'
+require 'net/http'
+
+set :port, 9494
+
+$users=[{name: 'foo', id: 1}]
+
+$delegate_user_call=false
+
+get "/v1/user" do
+  content_type :json
+  $users.to_json
+end
+
+get "/v1/user/:id" do
+  content_type :json
+  if ($delegate_user_call)
+    # call golang endpoint and return it
+    url = URI.parse('http://localhost:8080/v1/users/' + params[:id].to_s)
```

Comment:
This starts getting a bit weird. When you use + to concat a string, you end up with two outputs, and save the initial value. I would say this should be used, instead:

'http://localhost:8080/v1/users/' << params[:id]) 

Also, I believe the parameter hashes come in as Strings, anyhow. I could be wrong, as I'm going off of how Rails does things. ;)